include/seqan3/range/view/translation.hpp

```diff
@@ -33,256 +33,567 @@
-    */
-    template <typename irng_t>
-    friend auto operator|(irng_t && irange, seqan3::detail::translate_single_fn const & translate_single_fn_v)
+    /*!\name Element access
+     * \{
+     */
+    /*!\brief Return the n-th element.
+     * \param n The element to retrieve.
+     *
+     * Accessing an element behind the last causes undefined behaviour. In debug mode an assertion checks the size of
+     * the container.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (never modifies data).
+     */
+    reference operator[](size_type const n) const
     {
-        return translate_single_fn_v(std::forward<irng_t>(irange));
+        assert(n < size());
+        switch (data_members->tf)
+        {
+            case translation_frames::FWD_FRAME_0:
+                return translate_triplet((data_members->urange)[n * 3], (data_members->urange)[n * 3 + 1], (data_members->urange)[n * 3 + 2]);
+                break;
+            case translation_frames::REV_FRAME_0:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 1]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 2]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]));
+                break;
+            case translation_frames::FWD_FRAME_1:
+                return translate_triplet((data_members->urange)[n * 3 + 1], (data_members->urange)[n * 3 + 2], (data_members->urange)[n * 3 + 3]);
+                break;
+            case translation_frames::REV_FRAME_1:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 2]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 4]));
+                break;
+            case translation_frames::FWD_FRAME_2:
+                return translate_triplet((data_members->urange)[n * 3 + 2], (data_members->urange)[n * 3 + 3], (data_members->urange)[n * 3 + 4]);
+                break;
+            case translation_frames::REV_FRAME_2:
+                return translate_triplet(complement((data_members->urange)[(data_members->urange).size() - n * 3 - 3]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 4]), complement((data_members->urange)[(data_members->urange).size() - n * 3 - 5]));
+                break;
+            default:
+                throw std::invalid_argument("Invalid type of frame. Choose from FWD_FRAME_0, REV_FRAME_0, FWD_FRAME_1, REV_FRAME_1, FWD_FRAME_2 and REV_FRAME_2.");
+                break;
+        }
+    }
+    //!\}
+
+    //!\brief Implicit conversion to container types.
+    template <typename container_type>
+    operator container_type()
+    //!\cond
+        requires container_concept<std::decay_t<container_type>> &&
```

Comment:
if you are doing resize(), container_concept is not enough. `random_access_sequence_concept` would be accurate I think. Also you can move concept requirement to `template <foo_concept container_type>`